kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -141,6 +298,12 @@ enum State {
         this.virtualCluster = virtualCluster;
         this.logNetwork = virtualCluster.isLogNetwork();
         this.logFrames = virtualCluster.isLogFrames();
+        responsesByExceptionType = new ConcurrentHashMap<>();
```

Comment:
Does this need to be a CHM?  I think single threaded access is all that is required.